File: neet_code_150_cpp/appendix/nabeatu.cpp
#include <iostream>
#include <vector>

// 世界のナベアツのように3の倍数と3が含まれる数値のときに'aho'と出力する

// 3が含まれるか否かの判定
bool is_contain_3(int x) {
    while(x > 0) {
        if(x % 10 == 3) {
            return true;
        }
        x /= 10;
    }

    return false;
}

void nabeatu_array(const std::vector<int>& nums) {
    for(const auto& v: nums) {
        std::cout << v << " ";
        if(v % 3 == 0) {
            std::cout << "aho" << std::endl;
        } else if(is_contain_3(v)) {
            std::cout << "aho" << std::endl;
        } else {
            std::cout << std::endl;
        }
    }
}

void nabeatu_n(int n) {
    for(int i = 1; i <= n; ++i) {
        std::cout << i << " ";
        if(i % 3 == 0) {
            std::cout << "aho" << std::endl;
        } else if(is_contain_3(i)) {
            std::cout << "aho" << std::endl;
        } else {
            std::cout << std::endl;
        }
    }
}

int main(void) {
    auto case_1 = std::vector{1,2,3,4,5,6,7,8,9,10,11,12,13,14};
    //nabeatu_array(case_1);

    //nabeatu_n(20);
    nabeatu_n(40);
}

File: neet_code_150/src/appendix/nabeatu.rs
// 数字に3が含まれているか否かの判定
fn is_contain_3(mut x: i32) -> bool {
    while x > 0 {
        if x % 10 == 3 {
            return true;
        }

        x /= 10;
    }

    false
}

// 1 ~ n の間で3の倍数もしくは3が含まれる数字のときに'aho'と出力
fn nabeatu_n(n: i32) {
    for i in 1..=n {
        print!("{}: ",i);
        if i % 3 == 0 {
            println!("aho");
        } else if is_contain_3(i) {
            println!("aho");
        } else {
            println!("");
        }
    }
}

fn main() {
    nabeatu_n(100);
}
